### ISSUE TITLE
VirtualOutputSegment Write offset

### ISSUE THREAD

**Issue by Jimmys20** [dotnet/iot#1626]
Hello, in `Iot.Device.Multiplexing` the class`VirtualOutputSegment` has a method `void Write(byte value)` which states in the xml doc that the **_Least significant bit will be written to index 0._**

For example:
```
VirtualOutputSegment segment = new(8);
segment.Write(0b_1001_0110);

// Result:
// segment index: | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7
//         value: | 0 | 1 | 1 | 0 | 1 | 0 | 0 | 1
```

What I find odd is that when using a segment with bigger length that the written value the result is this:
```
VirtualOutputSegment segment = new(12);
segment.Write(0b_1001_0110);

// Result (offset to the left side of the byte value)
// segment index: | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11
//         value: | 0 | 0 | 0 | 0 | 0 | 1 | 1 | 0 | 1 | 0 |  0 |  1

// Expected result (offset to the right side of the byte value)
// segment index: | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11
//         value: | 0 | 1 | 1 | 0 | 1 | 0 | 0 | 1 | 0 | 0 |  0 |  0
```

To my understanding the least significant bit should always end up to index 0 and any possible offset should end up to the right. To explain with unit tests, I believe that the following two tests should be both be passing.

```
[Fact]
public void SegmentValuesWriteByte()
{
    VirtualOutputSegment segment = new(8);
    segment.Write(0b_1001_0110);

    Assert.True(
        segment[0] == 0 &&
        segment[1] == 1 &&
        segment[2] == 1 &&
        segment[3] == 0 &&
        segment[4] == 1 &&
        segment[5] == 0 &&
        segment[6] == 0 &&
        segment[7] == 1);
}

[Fact]
public void SegmentValuesWriteByteOffset()
{
    // segment length bigger than the written value
    VirtualOutputSegment segment = new(12);
    segment.Write(0b_1001_0110);

    Assert.True(
        segment[0] == 0 &&
        segment[1] == 1 &&
        segment[2] == 1 &&
        segment[3] == 0 &&
        segment[4] == 1 &&
        segment[5] == 0 &&
        segment[6] == 0 &&
        segment[7] == 1);
}
```

So is this possibly a bug or I didn't understand correct the way it should work?

**Comment by pgrawehr** [comment]
@richlander Probably your domain

**Comment by TheZlodziej** [comment]
It looks like the "offset" variable in Iot.Device.Multiplexing.Utility.VirtualOutputSegment.WriteByteAsValues(byte value, int offset) is unnecessary.

```
// calculate index by subtracting i from 7.
int index = 7 - i;
_values[index] = data;
```

**Comment by joperezr** [comment]
This is a great question. I believe your assumption above is right, and we do want those two unit tests you wrote to be passing. Sorry that I closed the PR #1662 which now I understand was trying to fix this other issue. Would you mind either re-opening that one, or a new one, and add the fix for this plus both of the unit tests above?
